apps/builder/app/builder/features/project-settings/section-marketplace.tsx

```diff
@@ -0,0 +1,322 @@
+import { useStore } from "@nanostores/react";
+import {
+  Grid,
+  InputField,
+  Label,
+  theme,
+  Text,
+  TextArea,
+  Button,
+  css,
+  Flex,
+  CheckboxAndLabel,
+  Checkbox,
+  InputErrorsTooltip,
+  PanelBanner,
+} from "@webstudio-is/design-system";
+import { ImageControl } from "./image-control";
+import {
+  $assets,
+  $authToken,
+  $marketplaceProduct,
+  $project,
+} from "~/shared/nano-states";
+import env from "~/shared/env";
+import { Image, createImageLoader } from "@webstudio-is/image";
+import { useIds } from "~/shared/form-utils";
+import { useEffect, useState } from "react";
+import { MarketplaceProduct } from "@webstudio-is/project-build";
+import { serverSyncStore } from "~/shared/sync";
+import { createTrpcRemixProxy } from "~/shared/remix/trpc-remix-proxy";
+import { type ProjectRouter } from "@webstudio-is/project";
+import { projectsPath } from "~/shared/router-utils";
+
+const imgStyle = css({
+  width: 72,
+  height: 72,
+  borderRadius: theme.borderRadius[4],
+  borderWidth: 1,
+  borderStyle: "solid",
+  borderColor: theme.colors.borderMain,
+});
+
+const defaultMarketplaceProduct: Partial<MarketplaceProduct> = {
+  category: "sectionTemplates",
+};
+
+const imageLoader = createImageLoader({
+  imageBaseUrl: env.IMAGE_BASE_URL,
+});
+
+const trpc = createTrpcRemixProxy<ProjectRouter>((method) =>
+  projectsPath(method, { authToken: $authToken.get() })
+);
+
+const validate = (data: MarketplaceProduct) => {
+  const parsedResult = MarketplaceProduct.safeParse(data);
+  if (parsedResult.success === false) {
+    return parsedResult.error.formErrors.fieldErrors;
+  }
+};
+
+const useMarketplaceApprovalStatus = () => {
+  const { send, data, state } = trpc.setMarketplaceApprovalStatus.useMutation();
+  const project = useStore($project);
+
+  const status =
+    data?.marketplaceApprovalStatus ??
+    project?.marketplaceApprovalStatus ??
+    "UNLISTED";
+
+  useEffect(() => {
```

Comment:
We can use onSuccess send callback without using `useEffect`
Also every mutation can have errors, what if it fails?

Reply:
OHH NICE, I forgot we have onSuccess and hated this effect while writing